src/client/datascience/insidersNativeNotebookSurveyBanner.ts

```diff
@@ -85,13 +85,17 @@ export class InsidersNativeNotebooksSurveyBanner implements IJupyterExtensionBan
     }
 
     public async showBanner(): Promise<void> {
+        if (this.disabledInCurrentSession) {
+            return;
+        }
+        // Disable for the current session.
+        this.disabledInCurrentSession = true;
         const executionCount: number = this.getExecutionCount();
         const notebookCount: number = this.getOpenNotebookCount();
-        const show = await this.shouldShowBanner(executionCount, notebookCount);
+        const show = this.shouldShowBanner(executionCount, notebookCount);
```

Comment:
I must be missing something. showShowBanner is still async.

Reply:
Damn.